File: amqp/receive.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');


amqp.connect('amqp://user:password@localhost', async (error0, connection) => {
  if (error0) {
    throw error0;
  }
  await connection.createChannel(function(error1, channel) {
    if (error1) {
      throw error1;
    }

    var queue = '';

    const exchange = 'events-discord';

    channel.assertExchange(exchange, 'fanout', '');

    channel.assertQueue(queue, {
      durable: false,
      exclusive: true
    }, (error2, q) => {
      if (error2) {
        throw error2;
      }
      console.log(" [*] Waiting for messages in %s. To exit press CTRL+C", q.queue);
      channel.bindQueue(q.queue, exchange, '');
      channel.consume(q.queue, (msg) => {
        console.log(" [x] %s", msg.content.toString());
      }, {
        noAck: true
      });
    });
  });
});

File: amqp/receive.py
#!/usr/bin/env python
import pika
import time
import sys

credentials = pika.PlainCredentials('user', 'password')
connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='localhost', credentials=credentials))
channel = connection.channel()

exchange = 'events-discord'
queue = 'sysA'

channel.exchange_declare(exchange=exchange, exchange_type='direct', durable=True)


result = channel.queue_declare(queue, exclusive=False)
queue_name = result.method.queue

channel.queue_bind(exchange=exchange, queue=queue_name)
print(' [*] %r binding on %r' % (exchange, queue_name))

print(' [*] Waiting for logs. To exit press CTRL+C')


def callback(ch, method, properties, body):
    # count += 1
    print(" [x] %r:%r" % (method.routing_key, body))
    time.sleep(1)
    channel.basic_ack(delivery_tag=method.delivery_tag)
    print(" [x] Ack message %r" % body)
    # print(" [x] Total %r" % count)


channel.basic_consume(
    queue=queue_name, on_message_callback=callback, auto_ack=False)

channel.start_consuming()
